include/seqan3/range/view/detail.hpp

```diff
@@ -60,6 +60,26 @@ namespace seqan3::detail
 struct view_base : public ranges::view_base
 {};
 
+// ============================================================================
+//  size_type_t_or_void
+// ============================================================================
+
+//!\brief TransformationTrait that resolves to seqan3::size_type_t if possible and void otherwise. [Default Overload]
+template <typename t>
+struct size_type_t_or_void
+{
+    //!\brief The default is to expose `void`.
+    using type = void;
+};
+
+//!\brief TransformationTrait that resolves to seqan3::size_type_t if possible and void otherwise. [Specialisation]
+template <std::ranges::SizedRange t>
+struct size_type_t_or_void<t>
```

Comment:
maybe this could simply be a mf alias with a conditional

Reply:
It would cause the same problem we are trying to solve with this metafunction, or not?